### PR TITLE
Removed some obsolete org.neo4j.helpers

### PR DESCRIPTION
**Pull request by lutovich**
1) removed `org.neo4j.helpers.collection.LruMap` because it was not used
2) removed `org.neo4j.helpers.Counter` because it was not used
3) removed `org.neo4j.helpers.SillyUtils` because it has duplicated functionality
4) removed `org.neo4j.helpers.Stats` because it is not used

### CODE REVIEW

File: community/kernel/src/main/java/org/neo4j/helpers/Counter.java [hunk 1]
```diff
@@ -1,39 +0,0 @@
...
- * GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License
- * along with this program.  If not, see <http://www.gnu.org/licenses/>.
- */
-package org.neo4j.helpers;
-
-import java.util.concurrent.atomic.AtomicLongFieldUpdater;
-
-public class Counter
```
Comment (chrisvest) [inline community/kernel/src/main/java/org/neo4j/helpers/Counter.java:24]: We cannot remove this without deprecating it first :(

File: community/kernel/src/main/java/org/neo4j/helpers/SillyUtils.java [hunk 2]
```diff
@@ -1,38 +0,0 @@
...
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License
- * along with this program.  If not, see <http://www.gnu.org/licenses/>.
- */
-package org.neo4j.helpers;
-
-/** Utility functions that are helpful for suppressing compiler warnings. */
-public class SillyUtils
```
Comment (chrisvest) [inline community/kernel/src/main/java/org/neo4j/helpers/SillyUtils.java:23]: We cannot remove this without deprecating it first :(

File: community/kernel/src/main/java/org/neo4j/helpers/Stats.java [hunk 3]
```diff
@@ -1,73 +0,0 @@
...
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License
- * along with this program.  If not, see <http://www.gnu.org/licenses/>.
- */
-package org.neo4j.helpers;
-
-public class Stats
```
Comment (chrisvest) [inline community/kernel/src/main/java/org/neo4j/helpers/Stats.java:22]: We cannot remove this without deprecating it first :(

File: community/kernel/src/main/java/org/neo4j/helpers/collection/LruMap.java [hunk 4]
```diff
@@ -1,39 +0,0 @@
...
- *
- * You should have received a copy of the GNU General Public License
- * along with this program.  If not, see <http://www.gnu.org/licenses/>.
- */
-package org.neo4j.helpers.collection;
-
-import java.util.LinkedHashMap;
-import java.util.Map;
-
-public class LruMap<K, V> extends LinkedHashMap<K,V >
```
Comment (chrisvest) [inline community/kernel/src/main/java/org/neo4j/helpers/collection/LruMap.java:25]: We cannot remove this without deprecating it first :(

**Comment by chrisvest** [comment]
@lutovich Removing the usages of these classes is fine, but the classes themselves cannot be removed because `org.neo4j.helpers` and `org.neo4j.helpers.collection` is unfortunately public API. You can instead deprecate everything in those packages.

**Comment by lutovich** [comment]
@chrisvest plan is to deprecate all removed classes in 2.3 once these PRs are in.

**Pull request merged**